plugins/inputs/cloud_pubsub/pubsub.go

```diff
@@ -170,30 +177,39 @@ func (ps *PubSub) startReceiver(parentCtx context.Context) error {
 
 // onMessage handles parsing and adding a received message to the accumulator.
 func (ps *PubSub) onMessage(ctx context.Context, msg message) error {
+	defer msg.Ack()
```

Comment:
`defer msg.Ack()` is not correct. This will run the ack no matter what exit path is taken from the function, but you instead want to be selective, only `ack`ing when the message has been properly handled, or is incorrectly formatted and you will not be able to ever process it correctly. Specifically when the context is cancelled you don't want the message to be `ack`ed

Reply:
Sounds good. I have reverted all the msg.Ack() changes.